src/components/common/Layout/components/Account/components/Notifications/MessagingSection.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Divider,
+  HStack,
+  Icon,
+  IconButton,
+  Link,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react"
+import { XMTPProvider, useCanMessage } from "@xmtp/react-sdk"
+import Button from "components/common/Button"
+import { Modal } from "components/common/Modal"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import useSubmit from "hooks/useSubmit"
+import dynamic from "next/dynamic"
+import { ArrowSquareOut, Gear } from "phosphor-react"
+import { useEffect } from "react"
+import { useAccount } from "wagmi"
+import SubscriptionPromptSkeleton from "../MessageSkeleton/SubscriptionPromptSkeleton"
+import { SubscriptionPrompt } from "./SubscriptionPrompt"
+import NotificationsSection from "./components/NotificationsSection"
+import useWeb3InboxSubscription from "./components/useWeb3InboxSubscription"
+import { useSubscribeToXMTP } from "./hooks/useSubscribeToXMTP"
+
+const Messages = dynamic(() => import("./Messages"))
+
+const MessagingSection = () => {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const {
+    isRegistering,
+    isSigning,
+    isSubscribing,
+    isLoading,
+    subscriptionStatus,
+    subscribe,
+  } = useWeb3InboxSubscription()
+  const { canMessageStatic } = useCanMessage()
+  const showErrorToast = useShowErrorToast()
+
+  const { address } = useAccount()
+
+  const {
+    isLoading: isCheckingXMTPAccess,
```

Comment:
I'd call these `isCanMessageLoading`, `onCanMessageSubmit` (but you don't need this one, check my next comment!), `canMessage`

Reply:
I think "isCheckingXMTPAccess" accurately describes its purpose.  I think "CanMessageLoading" doesn't specify whether it's for XMTP or W3I.